### DISCUSSION TITLE
Default logger with multiple targets controlled by environment variables

### DISCUSSION THREAD

**Discussion by AndyDralle** [gabime/spdlog#3005] [Q&A] [answered]
Hey there!

I have taken a look at https://github.com/gabime/spdlog/discussions/2444 and https://github.com/gabime/spdlog/issues/79. While very much related to my issue, I don't think they solve it completely, though: 

I would like to use the `default_logger` for logging to multiple targets, such as a file and stdout. Further, I would like to use `spdlog::cfg::load_env_levels()` to control the logging to these different targets. My goal is to set the log level individually using separate environment variables for each target. 

Approach A (https://github.com/gabime/spdlog/issues/79):
While I could create a logger with multiple sinks, each with their own log level, I can't control the level of a sink via an environment variable. This only works for loggers.

Approach B (https://github.com/gabime/spdlog/discussions/2444):
I don't actually know how to create a sink that forwards messages to multiple loggers. 

Say I have a list of loggers: `std::vector<spdlog::logger> loggers`. 
If I implement `sink_it_` for my custom sink like this: 
```
  void sink_it_(const spdlog::details::log_msg& message) override {
    for (auto& logger : loggers) {
      logger.sink_it_(message);
    }
  }
```
I skip the actual logging part. I.e., by directly sinking, I skip the logic verifying the log level, right? 
I don't know how to call `.log()` on my loggers, given that my message is of type `log_msg&`. 

Is there a way to achieve this?

**Comment by gabime** [comment, marked as answer]
Try passing the `message.level` and `message.payload` to each logger.log()

**Reply by AndyDralle** [reply]
That seems to do the trick. My sink implementation now looks like this: 
```C++
  void sink_it_(const spdlog::details::log_msg& message) override {
    for (auto& logger : loggers) {
      logger->log(message.time, message.source, message.level, message.payload);
    }
  }

  void flush_() override {
    for (auto& logger : loggers) {
      logger->flush();
    }
  }
```

Thank you!
